bootstrap/kubeadm/api/v1alpha4/condition_consts.go

```diff
@@ -51,6 +51,17 @@ const (
 	DataSecretGenerationFailedReason = "DataSecretGenerationFailed"
 )
 
+const (
+	// IgnitionGenerated documents a status of the bootstrap data generation process
+	// in Ignition format.
+	IgnitionGeneratedCondition = "IgnitionGeneratedCondition"
```

Comment:
I don't think we should add ignition-specific conditions. What about using DataSecretAvailable + a new reason, if required

Reply:
Hm, `DataSecretAvailable` is a boolean, which will be set anyway IIUC. I don't think we should be setting it to false if there are warnings.

Reply:
> DataSecretAvailable is a boolean

DataSecretAvailable is a condition, so it can be false and having a reson explaining why it is false

Reply:
But, if it's false, doesn't that mean the data is _not_ available, which wouldn't be true in this case?

Reply:
That's is a problem if we are trying to use condition to convey warning collected during processing...
Condition should represent current operational state, so if DataSecretAvailable=true, no message.